### PR TITLE
ENCD-5678-update-frontend-documentation

### PR DESCRIPTION
**Pull request by zoldello**

### CODE REVIEW

File: docs/front-end-overview.md [hunk 1]
```diff
@@ -183,3 +183,96 @@ The part within the callback would execute once the GET request had returned, wi
...
+
+## ESLint
+ENCODE uses [ESLint](https://eslint.org/) to automate enforcement of Javascript code quality and standards. It is also a tool that helps developers learn good Javascript practices. ESLint inspects all Javascript code in:
+
+`src/encoded/static/components/`
+
+ENCODE implementation of [ESLint](https://eslint.org/) uses [AirBnB](https://github.com/airbnb/javascript) coding style; although some exceptions are made to the rules. The rules can be found by at [https://github.com/ENCODE-DCC/encoded/blob/dev/.eslintrc.js](https://github.com/ENCODE-DCC/encoded/blob/dev/.eslintrc.js),  under the "Rules" section. The rules should not be disabled (via eslint.js or in a code file) without first consulting other front-end developers. There is rarely ever a need to disable a rule for a ticket and almost always there are legitimate workarounds.
+
+It is important to make sure new Javascript code satisifies [ESLint](https://eslint.org/) rules, otherwise, the CircleCI under  (ci/circleci: eslint) check will fail the build.
```
Comment (emro) [inline docs/front-end-overview.md]: Extra space after "under"?

File: docs/front-end-overview.md [hunk 2]
```diff
@@ -183,3 +183,96 @@ The part within the callback would execute once the GET request had returned, wi
...
+
+You can check for and correct [ESLint](https://eslint.org/) violations either through the command line or with a plugin to a text editor. Either way suffices as long  as violations are corrected.
```
Comment (emro) [inline docs/front-end-overview.md]: Extra space after "long"?

File: docs/front-end-overview.md [hunk 3]
```diff
@@ -183,3 +183,96 @@ The part within the callback would execute once the GET request had returned, wi
...
+
+### Command line
+
+#### Determining [ESLint](https://eslint.org/)  Violations
+
+The  format for checking for [ESLint](https://eslint.org/) violations is:
+`./node_modules/eslint/bin/eslint.js <file or folder path>`
+
+for example, to get all rule violations in a folder, you can run:
+`./node_modules/eslint/bin/eslint.js src/encoded/static/components/`
```
Comment (emro) [inline docs/front-end-overview.md:206]: I'd capitalize "for" because you do in line 216.

File: docs/front-end-overview.md [hunk 4]
```diff
@@ -183,3 +183,96 @@ The part within the callback would execute once the GET request had returned, wi
...
+
+#### Correcting [ESLint](https://eslint.org/) Violations
+
+The format for correcting [ESLint](https://eslint.org/) violations is:
+`./node_modules/eslint/bin/eslint.js <file or folder path> --fix`
+
+For example, you fix all rule violations in a folder, by running:
+`./node_modules/eslint/bin/eslint.js src/encoded/static/components/ --fix`
+
+You can also fix violation for just one file (e.g datacolor.js) by  running:
```
Comment (emro) [inline docs/front-end-overview.md]: Extra space after "by"?

File: docs/front-end-overview.md [hunk 5]
```diff
@@ -183,3 +183,96 @@ The part within the callback would execute once the GET request had returned, wi
...
+`./node_modules/eslint/bin/eslint.js src/encoded/static/components/datacolors.js -- fix`
+
+Numerous other options can be found at [ESLint Command line interface Documention page](https://eslint.org/docs/user-guide/command-line-interface)
+
+There are many [YouTube](https://www.youtube.com/results?search_query=eslint+command+line) videos on this  topic. Here is one [discussing command line](https://www.youtube.com/watch?v=txJEn1Fn0Gg).
```
Comment (emro) [inline docs/front-end-overview.md]: Extra space after "this"?

File: docs/front-end-overview.md [hunk 6]
```diff
@@ -183,3 +183,96 @@ The part within the callback would execute once the GET request had returned, wi
...
+
+### Editor Plugin
+
+Widely used code text editors generally have extensions that allows you to visually detect [ESLint](https://eslint.org/) violations by seeing a squiggly under the code in question. They also display  information on what rule was violated by mousing over the squiggly and sometimes automated correction by right clicking and selecting an appropiate option.
```
Comment (emro) [inline docs/front-end-overview.md]: "that allow" not "that allows"

Comment (emro) [reply docs/front-end-overview.md]: Extra space after "display"

Comment (emro) [reply docs/front-end-overview.md]: This is sort of a confusing sentence though I do know what you mean. Maybe "Editors display information pertaining to the violated rule on hover as well as automated corrections which can be selected and applied by clicking.

File: docs/front-end-overview.md [hunk 7]
```diff
@@ -183,3 +183,96 @@ The part within the callback would execute once the GET request had returned, wi
...
+
+## StyleLint
+ENCODE uses [StyleLint](https://stylelint.io/) to automate enforcement of proper style sheets- CSS/SCSS (henceforth called  just CSS) format and structure. It is also a good tool that helps designers learn good CSS practices. StyleLint inspects all style sheets in:
+
+`src/encoded/static/scss/`
+
+ENCODE implementation of [StyleLint](https://stylelint.io/)  uses [Stylelint Config Standard](https://github.com/stylelint/stylelint-config-standard) rules collection; although some exceptions are made to the rules. The rules can  be found by examining [https://github.com/ENCODE-DCC/encoded/blob/dev/.stylelint.json](https://github.com/ENCODE-DCC/encoded/blob/dev/.stylelint.json),  under the "Rules" section. The rules should not be disabled without consulting other front-end developers. There is rarely ever a need to disable any rule and almost always there are legitimate workarounds. The current disabled rules are [design debt](https://en.wikipedia.org/wiki/Technical_debt).
+
+It is important to make sure new style sheets  satisifies [StyleLint](https://stylelint.io/) rules, otherwise, the CircleCI under  (ci/circleci: stylelint) check will fail the build.
```
Comment (emro) [inline docs/front-end-overview.md]: Extra space after "sheets" and "satisfies" should be "satisfy"

Comment (emro) [reply docs/front-end-overview.md]: Extra space after "under"

File: docs/front-end-overview.md [hunk 8]
```diff
@@ -183,3 +183,96 @@ The part within the callback would execute once the GET request had returned, wi
...
+
+You can check for and correct [StyleLint](https://stylelint.io/) violations either through the command line or with a plugin to a text editor. Either way suffices as long  as violations are corrected.
+
```
Comment (emro) [inline docs/front-end-overview.md:244]: Extra space after "long"

File: docs/front-end-overview.md [hunk 9]
```diff
@@ -183,3 +183,96 @@ The part within the callback would execute once the GET request had returned, wi
...
+### Command line
+
+#### Determining [StyleLint](https://stylelint.io/)  Violations
+
+The  format for checking for [StyleLint](https://stylelint.io/) violations is:
```
Comment (emro) [inline docs/front-end-overview.md]: Extra space after "the"

File: docs/front-end-overview.md [hunk 10]
```diff
@@ -183,3 +183,96 @@ The part within the callback would execute once the GET request had returned, wi
...
+You can also check just one file (e.g datacolor.js) by  running:
+`./node_modules/stylelint/bin/eslint.js src/encoded/static/scss/style.scss`
+
+#### Correcting [StyleLint](https://stylelint.io/) Violations
+
+The format for correcting the [StyleLint](https://stylelint.io/) violations is:
+`./node_modules/stylelint/bin/stylelint.js <file or folder path> --fix`
+
+For example, you fix all rule violations in a folder, by running:
+`./node_modules/stylelint/bin/stylelint.js src/encoded/static/scss/ --fix`
```
Comment (emro) [inline docs/front-end-overview.md:264]: No comma after "folder"

File: docs/front-end-overview.md [hunk 11]
```diff
@@ -183,3 +183,96 @@ The part within the callback would execute once the GET request had returned, wi
...
+
+You can also fix violation for just one file (e.g style.css) by  running:
+`./node_modules/stylelint/bin/stylelint.js src/encoded/static/components/style.scss -- fix`
```
Comment (emro) [inline docs/front-end-overview.md:267]: Extra space

File: docs/front-end-overview.md [hunk 12]
```diff
@@ -183,3 +183,96 @@ The part within the callback would execute once the GET request had returned, wi
...
+
+Numerous other options can be found at [StyleLint Command line interface Documention page](https://stylelint.io/user-guide/usage/cli)
+
+You can search on [YouTube](https://www.youtube.com/results?search_query=stylelint+) to see if there are any videos that illustrate using the command line with  [StyleLint](https://stylelint.io/).  In general, the idea is similar to  the [ESLint](https://eslint.org/).
```
Comment (emro) [inline docs/front-end-overview.md]: Extra spaces

File: docs/front-end-overview.md [hunk 13]
```diff
@@ -183,3 +183,96 @@ The part within the callback would execute once the GET request had returned, wi
...
+#### Editor Extension
+
+Widely used code text editors generally have extensions that allows you to visually detect [StyleLint Command line interface Documention page](https://stylelint.io/user-guide/usage/cli) violations by seeing a squiggly under the code in question. They also display information on what rule was violated by mousing over the squiggly and sometimes automated correction by right clicking and selecting an appropiate option.
```
Comment (emro) [inline docs/front-end-overview.md]: "allows" should be "allow"

Comment (emro) [reply docs/front-end-overview.md]: "allow you to detect by seeing" is sort of wordy construction, you could just say "Widely used code text editors have extensions that highlight code violations with underlines/emphasis, and often display information about rule violations and sometimes automated corrections on mouseover...."

Comment (emro) [reply docs/front-end-overview.md]: "appropriate" not "appropiate"

File: docs/front-end-overview.md [hunk 14]
```diff
@@ -183,3 +183,96 @@ The part within the callback would execute once the GET request had returned, wi
...
+
+Here is collection of many extenstions for [popular text editors](https://stylelint.io/user-guide/integrations/editor)
+
+You can search on [YouTube](https://www.youtube.com/results?search_query=stylelint+) to see if there are any vidoes that illustrate using text editor extension with  [StyleLint](https://stylelint.io/). In general, the idea is similar to  the [ESLint](https://eslint.org/).
```
Comment (emro) [inline docs/front-end-overview.md]: Extra space after "with"

**Review by emro — CHANGES_REQUESTED** [review]
There were a number of extra spaces which should be omitted. I have some wording suggestions which you should feel free to ignore. A couple typos should be fixed. It looks good though! All the links will be very helpful.

**Comment by zoldello** [comment]
@emro I will fix all double-spaces. However, in general, markdown files ignores the extra space just like HTML. It may be easier if you use a markdown reader. Popular code editors have extensions that show you the markdown code and generated document side-by-side.

File: docs/front-end-overview.md [hunk 15]
```diff
@@ -185,94 +185,94 @@ The part within the callback would execute once the GET request had returned, wi
 You’ll see some uses of these request functions that _do_ fit into a “get data, render data” model. Earlier versions of _encoded_ had a strict limit on the length of a URL. In cases where our search URI could have a long length, you had to use these request functions because if a URI’s length reached a certain maximum, these functions would break the request into _multiple_ GET requests so that each URI fit into the maximum. Once all the GET requests returned, these functions put the results together and returned them to the caller. Now that _encoded_ accepts arbitrarily long URIs, these cases could be replaced with `<FetchedData>`.
 
 ## ESLint
-ENCODE uses [ESLint](https://eslint.org/) to automate enforcement of Javascript code quality and standards. It is also a tool that helps developers learn good Javascript practices. ESLint inspects all Javascript code in:
+ENCODE uses [ESLint](https://eslint.org/) to automate enforcement of Javascript code quality and standards. It is also a tool that help developers learn good Javascript practices. ESLint inspects all Javascript code in:
```
Comment (emro) [inline docs/front-end-overview.md]: This should be "helps" actually.

File: docs/front-end-overview.md [hunk 16]
```diff
@@ -185,94 +185,94 @@ The part within the callback would execute once the GET request had returned, wi
...
 #### Editor Extension
 
-Widely used code text editors generally have extensions that allows you to visually detect [StyleLint Command line interface Documention page](https://stylelint.io/user-guide/usage/cli) violations by seeing a squiggly under the code in question. They also display information on what rule was violated by mousing over the squiggly and sometimes automated correction by right clicking and selecting an appropiate option.
+Popular code Editors display information pertaining to [StyleLint Command line interface Documention page](https://stylelint.io/user-guide/usage/cli) violated rule on hover as well as automated corrections, which can be selected and applied by clicking on fix (wording may vary depending on editor).
 
-Here is collection of many extenstions for [popular text editors](https://stylelint.io/user-guide/integrations/editor)
+Here is a collection of extenstions for [popular code editors](https://stylelint.io/user-guide/integrations/editor)
```
Comment (emro) [inline docs/front-end-overview.md]: I think I missed this in the first review - sorry! - "extenstions" should be "extensions". Also in the text above which you did not change "Documention" should be "Documentation".

**Review by emro — CHANGES_REQUESTED** [review]
Two more small changes! Thank you for making all the other edits.

**Review by emro — APPROVED** [review]

**Pull request merged**